### ISSUE TITLE
Inconsistent behaviour with abstract classes when decorated with @dataclass.

### ISSUE THREAD

**Issue by jkb0o** [microsoft/pyright#2331]
`pyright test.py` do not warn me with `Cannot instantiate abstract class "Concreete"` with this snapshot:
```python
# test.py
from abc import ABCMeta, abstractmethod
from dataclasses import dataclass

class Abstract(metaclass=ABCMeta):
    @abstractmethod
    def test(self): ...

@dataclass
class Concreete(Abstract):
    pass

c = Concreete()
```
If I comment out the `@dataclass` line, I get the expected warning.
If I use `@dataclass()` instead of `@dataclass`, I also get expected warning.
Executing the code ends with exception `Can't instantiate abstract class Concreete with abstract method test` in any case.

**Expected behavior**
I should be warned with error in both `@dataclass` and `@dataclass()` cases.

**VS Code extension or command-line**
Reproducible with pyright and pylance both. I've tested with `pyright==1.1.170`

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

In the meantime, you can work around the problem by using the call form of the `dataclass` decorator, like this:

```python
@dataclass()
class Concreete(Abstract):
    pass
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.171 which was just published. It will also be included in the next release of pylance.
